src/foam/dao/index/AltIndex.js

```diff
@@ -32,21 +41,79 @@ foam.CLASS({
 
   properties: [
     {
+      /** delegate factories */
       name: 'delegates',
       factory: function() { return []; }
-    }
+    },
+    {
+      /** factory quick lookup */
+      name: 'delegateMap_',
+      factory: function() { return {}; }
+    },
   ],
 
   methods: [
-    function addIndex(index) {
-      // Populate the index
-      var a = foam.dao.ArraySink.create();
-      this.plan(a).execute([Promise.resolve()], a);
 
-      index.bulkLoad(a);
-      this.delegates.push(index);
+    /** Returns smallest estimate from the delegates */
+    function estimate(size, sink, skip, limit, order, predicate) {
```

Comment:
estimate() is the 'static' version of plan, that estimates performance without having any real data. Estimates should be relatively accurate when compared to each other, but won't be directly comparable to plan.cost since that will be much more accurate.